File: 프로그래머스/LV1/코딩테스트 연습/연습문제/기사단원의 무기/기사단원의 무기.py
def get_number_of_divisors(n):
    ret = 0
    for i in range(1, int(n**0.5) + 1):
        if i*i == n:
            ret += 1
        elif n % i == 0:
            ret += 2

    return ret

def solution(number, limit, power):
    powers = [get_number_of_divisors(i) for i in range(1,number+1)]
    return sum([power if p > limit else p for p in powers])

File: 프로그래머스/LV1/코딩테스트 연습/연습문제/기사단원의 무기/기사단원의 무기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>
int get_number_of_divisors(int n)
{
    int ret = 0;

    for(int i = 1; i <= (int)pow(n, 0.5); i++)
    {
        if(i*i == n)ret++;
        else if(n % i == 0)ret += 2;
    }
    return ret;
}

int solution(int number, int limit, int power) 
{
    int answer = 0;
    int p;
    for(int i = 1; i <= number; i++)
    {
        p = get_number_of_divisors(i);
        answer += p > limit ? power : p; 
    }
        
    return answer;
}